Python/ast.c

```diff
@@ -3146,7 +3146,7 @@ ast_for_call(struct compiling *c, const node *n, expr_ty func,
             }
             else if (TYPE(CHILD(ch, 1)) == comp_for) {
                 /* the lone generator expression */
-                e = copy_location(ast_for_genexp(c, ch), maybegenbeg);
+                e = copy_location(ast_for_genexp(c, ch), maybegenbeg, closepar);
```

Comment:
Note that if `maybegenbeg` and `closepar` are passed here and used as start and end line/col info we will need to work around this in pegen, since the parser there does not consider the parentheses part of the generator expression and uses its first and last character for the node metadata.

Reply:
Clearly this is the intention of the original code -- it started out pointing the start location to the open parenthesis. This is apparently so that the location for a genexp node *always* includes the surrounding parentheses -- which are considered part of the syntax (just like for a list comprehension the `[...]` are considered part of the syntax).

There is some discussion of this in bpo-31241 (though not for function calls).

We can probably handle this in pegen by writing a separate bit of function call syntax, e.g.
```primary:
    | ...
    | primary lone_genexp
    | primary '(' [arguments] ')'
    | ...
lone_genexp:
    | '(' expression for_if_clauses ')'
```
(We'll need a similar thing for `call_tail` in the targets section.)